pkg/activator/handler/handler.go

```diff
@@ -34,19 +33,36 @@ type ActivationHandler struct {
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
 	Reporter  activator.StatsReporter
+	Throttler *activator.Throttler
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
-
 	start := time.Now()
+	revID := activator.RevisionID{namespace, name}
+
+	// ActiveEndpoint() will block until the first endpoint is available.
+	ar := a.Activator.ActiveEndpoint(namespace, name)
```

Comment:
Let's handle the error of the ActivationRequest in this function rather than passing it down?